### PR TITLE
Use justinmimbs/date for internal representation

### PR DESCRIPTION
**Pull request by jorgengranseth**
This package saves us a lot of effort by defining many of the functions we use. It even has a `today` function to give us the current date in the user's local time zone. 👌

I suggest merging this into the 0.19-branch and see which options we have for API improvements based on this.

Opinions?

### CODE REVIEW

File: elm.json [hunk 1]
```diff
@@ -12,7 +12,8 @@
         "elm/core": "1.0.0 <= v < 2.0.0",
         "elm/html": "1.0.0 <= v < 2.0.0",
         "elm/json": "1.0.0 <= v < 2.0.0",
-        "justinmimbs/date": "1.1.0 <= v < 2.0.0"
+        "elm/time": "1.0.0 <= v < 2.0.0",
+        "justinmimbs/date": "2.0.1 <= v < 3.0.0"
```
Comment (nelsonic) [inline elm.json]: https://github.com/justinmimbs/date/tree/2.0.1 (_for anyone following along ..._)

**Comment by jorgengranseth** [comment]
Any idea how to get Cypress to run on the build server?
https://travis-ci.org/dwyl/elm-datepicker/jobs/650880948#L314

If that has never worked I can remove it - just seemed useful to auto-run considering the tests were broken.

**Comment by nelsonic** [comment]
@jorgengranseth do the Cypress tests run on your `localhost`?

**Comment by jorgengranseth** [comment]
@nelsonic yeah, works fine with `npm install` and `npm test` on this branch. `rm -rf elm-stuff` if you run into trouble. Elm-test broke for me having also run on `master` due to  folder structure in `elm-stuff` (legacy 0.18 seems to break it).

File: .travis.yml [hunk 2]
```diff
@@ -3,6 +3,12 @@ sudo: false # don't need root access on Travis-CI (builds faster without sudo!)
 language: node_js # elm is installed from npm (see install below)
 node_js: node     # use "latest" version of Node.js
 
+# See https://github.com/cypress-io/cypress-example-kitchensink/blob/master/.travis.yml
+addons:
+  apt:
+    packages:
+      - libgconf-2-4
```
Comment (nelsonic) [inline .travis.yml]: 👍

**Comment by nelsonic** [comment]
@jorgengranseth looks like you've got the Cypress tests running on Travis-CI 🎉 
https://travis-ci.org/dwyl/elm-datepicker/builds/650885256#L309
image

**Pull request merged**